### ISSUE TITLE
Warning if property "watch" does not match "Object.watch" signature

### ISSUE THREAD

**Issue by syranide** [google/closure-compiler#2517]
While researching #2516 I came across this. It doesn't make sense to me at least. Overriding `watch` causes a warning about the signature being incorrect. Overriding other `Object` properties generally doesn't seem to cause this warning.

```js
WARNING - Returned type does not match declared return type.
Expected : undefined
Found : string
(new (class {watch() { return "watch"; }})).watch();
```
```
java -jar compiler.jar -O ADVANCED --js in.js --js_output_file out.js --language_in ES6_STRICT --language_out ES5_STRICT --new_type_inf --env BROWSER
```

**Comment by brad4d** [comment]
I believe this is working as intended.

The default externs say that `Object.prototype.watch` is supposed to return undefined.
https://github.com/google/closure-compiler/blob/master/externs/es3.js#L467

It makes sense to give a warning if you override this behavior, since some code might rely on watch returning `undefined`. The warning may go away if you explicitly give JSDoc for the method specifying what the return type should be.

```javascript
class C {
  /**
   * @override
   * @return {string}
   */
  watch() { return "watch"; }
}
```
